docs/source/serving.rst

```diff
@@ -1,12 +1,468 @@
+
+    # Manual testing
+    prompt = "Hey, are you conscious? Can you talk to me?"
+    messages = [
+        {
+            "role": "system",
+            "content": "",
+        },
+        {"role": "user", "content": prompt},
+    ]
+    templated_prompt = tokenizer.apply_chat_template(
+        messages,
+        tokenize=False,
+        add_generation_prompt=True,
+    )
+    print("Prompt:", prompt)
+    print("Templated prompt:", templated_prompt)
+    inputs = tokenizer(
+        templated_prompt,
+        return_tensors="pt",
+    ).to("cuda")
+    generated_ids = quantized_model.generate(**inputs, max_new_tokens=128)
+    output_text = tokenizer.batch_decode(
+        generated_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False
+    )
+    print("Response:", output_text[0][len(prompt):])
+
+
+Step 3: Export to ExecuTorch
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+    # Install ExecuTorch
+    git clone https://github.com/pytorch/executorch.git
+    cd executorch
+    ./install_requirements.sh
+
+    # Convert checkpoint format for ExecuTorch
+    python -m executorch.examples.models.phi_4_mini.convert_weights pytorch_model.bin pytorch_model_converted.bin
+
+    # Export to PTE format with torchao optimizations preserved
+    PARAMS="executorch/examples/models/phi_4_mini/config.json"
+    python -m executorch.examples.models.llama.export_llama \
+        --model "phi_4_mini" \
+        --checkpoint "pytorch_model_converted.bin" \
+        --params "$PARAMS" \
+        -kv \
+        --use_sdpa_with_kv_cache \
+        -X \
+        --metadata '{"get_bos_id":199999, "get_eos_ids":[200020,199999]}' \
+        --max_seq_length 128 \
+        --max_context_length 128 \
+        --output_name="phi4-mini-8da4w.pte"
+
+
+Mobile Performance Characteristics
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The torchao-optimized 8da4w model provides:
+
+- **Memory**: ~3.2GB on iPhone 15 Pro
+- **Speed**: ~17 tokens/sec on iPhone 15 Pro
+- **Accuracy**: Maintained within 5-10% of original model on most benchmarks
+
+.. note::
+    For detailed instructions on testing the executorch model and reproducing benchmarks please refer to the `HF Phi-4-mini-instruct-8da4w model <https://huggingface.co/pytorch/Phi-4-mini-instruct-8da4w>`_.
+
+Evaluation
+---------
+
+Model Quality Assessment
+^^^^^^^^^^^^^^^^^^^^^^
+
+Evaluate quantized models using lm-evaluation-harness:
+
+.. code-block:: bash
+
+    # Install evaluation framework
+    # Need to install lm-eval from source: https://github.com/EleutherAI/lm-evaluation-harness#install
+
+    # Evaluate baseline model
+    lm_eval --model hf --model_args pretrained=microsoft/Phi-4-mini-instruct --tasks hellaswag --device cuda:0 --batch_size 8
+
+    # Evaluate torchao-quantized model (float8dq)
+    lm_eval --model hf --model_args pretrained=pytorch/Phi-4-mini-instruct-float8dq --tasks hellaswag --device cuda:0 --batch_size 8
+
+Memory Benchmarking
+^^^^^^^^^^^^^^^^^
+
```

Comment:
Should add some text here. "For the Phi-4-mini-instruct model, serving with float8 dynamic quantization used X% less memory" or something